algoworld_contracts/swapper/swap_proxy.py

```diff
@@ -0,0 +1,119 @@
+"""
+MIT License
+
+Copyright (c) 2022 AlgoWorld
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import dataclasses
+import sys
+
+from pyteal import (
+    Addr,
+    And,
+    Balance,
+    Bytes,
+    Cond,
+    Global,
+    Gtxn,
+    Int,
+    Mode,
+    Or,
+    Substring,
+    Txn,
+    TxnType,
+    compileTeal,
+)
+
+from algoworld_contracts.common.utils import parse_params
+
+"""
+Swapper Proxy Used for Storing Swap Configurations
+1. Activate Proxy
+2. Store Swap Configurations
+"""
+
+TEAL_VERSION = 6
+
+STORE_GSIZE = Int(2)
+STORE_FEE = 0
+STORE_PROXY_NOTE = 1
+
+
+@dataclasses.dataclass
+class SwapProxy:
+    swap_creator: str
+    version: str
+
+
+def swapper_proxy(cfg: SwapProxy):
+
+    is_proxy_store = And(
+        Global.group_size() == STORE_GSIZE,
+        Gtxn[STORE_FEE].type_enum() == TxnType.Payment,
+        Gtxn[STORE_PROXY_NOTE].type_enum() == TxnType.Payment,
+    )
+
+    return Cond([is_proxy_store, proxy_store(cfg)])
+
+
+def proxy_store(cfg: SwapProxy):
+    Balance(Txn.sender())
+
+    store_fee = And(
+        Gtxn[STORE_FEE].type_enum() == TxnType.Payment,
+        Or(
+            Gtxn[STORE_FEE].amount() == Int(110_000),
+            Gtxn[STORE_FEE].amount() == Int(10_000),
+        ),
+        Gtxn[STORE_FEE].sender() == Addr(cfg.swap_creator),
+        Gtxn[STORE_FEE].receiver() == Txn.sender(),
```

Comment:
I'm not sure about what is happening here: can you give a description of this step? Who is paying who, for what? Who is executing the `STORE_FEE` transaction? Is that a user or the Smart Contract?

Reply:
Perhaps `Txn.sender(),` is not used in the right context here but this transaction is the one that creator of the proxy pays to cover the fee for the second transaction that issued on behalf of the escrow (with logisig) that sends 0 algos to itself and puts ipfs note in there. 

With `Gtxn[STORE_FEE].receiver() == Txn.sender()` the intent was to check if the receiver is the escrow account, since tests werent failing i assume  Txn.sender() in this context does indeed return me the address of the escrow?

Reply:
Also, pushed a commit that removes them already